test/tests/diff.js

```diff
@@ -91,58 +91,101 @@ describe("Diff", function() {
-    assert.equal(patch.size(), 1);
-    assert.ok(patch.isModified());
-
-    var hunk = patch.hunks()[0];
-    assert.equal(hunk.size(), 5);
-
-    var lines = hunk.lines();
-    assert.equal(lines[0].origin(), Diff.LINE.CONTEXT);
-    assert.equal(lines[1].origin(), Diff.LINE.CONTEXT);
-    assert.equal(lines[2].origin(), Diff.LINE.CONTEXT);
-
-    var oldContent = "__Before submitting a pull request, please ensure " +
-      "both unit tests and lint checks pass.__\n";
-    assert.equal(lines[3].rawContent(), oldContent);
-    assert.equal(lines[3].origin(), Diff.LINE.DELETION);
-    assert.equal(lines[3].contentLen(), 90);
-
-    var newContent = "__Before submitting a pull request, please ensure " +
-      "both that you've added unit tests to cover your shiny new code, " +
-      "and that all unit tests and lint checks pass.__\n";
-    assert.equal(lines[4].rawContent(), newContent);
-    assert.equal(lines[4].origin(), Diff.LINE.ADDITION);
-    assert.equal(lines[4].contentLen(), 162);
+    return this.diff[0].patches()
+      .then(function(patches) {
+        var patch = patches[0];
+
+        assert.equal(patch.oldFile().path(), "README.md");
+        assert.equal(patch.newFile().path(), "README.md");
+        assert.equal(patch.size(), 1);
+        assert.ok(patch.isModified());
+
+        return patch.hunks();
+      })
+      .then(function(hunks) {
+        var hunk = hunks[0];
+        assert.equal(hunk.size(), 5);
+
+        return hunk.lines();
+      })
+      .then(function(lines) {
+        assert.equal(lines[0].origin(), Diff.LINE.CONTEXT);
+        assert.equal(lines[1].origin(), Diff.LINE.CONTEXT);
+        assert.equal(lines[2].origin(), Diff.LINE.CONTEXT);
+
+        var oldContent = "__Before submitting a pull request, please ensure " +
+          "both unit tests and lint checks pass.__\n";
+        assert.equal(lines[3].rawContent(), oldContent);
+        assert.equal(lines[3].origin(), Diff.LINE.DELETION);
+        assert.equal(lines[3].contentLen(), 90);
+
+        var newContent = "__Before submitting a pull request, please ensure " +
+          "both that you've added unit tests to cover your shiny new code, " +
+          "and that all unit tests and lint checks pass.__\n";
+        assert.equal(lines[4].rawContent(), newContent);
+        assert.equal(lines[4].origin(), Diff.LINE.ADDITION);
+        assert.equal(lines[4].contentLen(), 162);
+      });
   });
 
   it("can diff the workdir with index", function() {
-    var patches = this.workdirDiff.patches();
-    assert.equal(patches.length, 3);
-    assert(patches[2].isUntracked());
+    return this.workdirDiff.patches()
+      .then(function(patches) {
+        assert.equal(patches.length, 3);
+        assert(patches[2].isUntracked());
 
-    var oldFile = patches[2].delta.oldFile();
-    assert.equal(oldFile.path(), "wddiff.txt");
-    assert.equal(oldFile.size(), 0);
+        var oldFile = patches[2].delta.oldFile();
```

Comment:
Hmm.. so after looking at the diff I'm not sure why those are highlighted. They aren't tabs in the raw file nor in my editor.